src/java.base/share/classes/sun/security/provider/SHA3.java

```diff
@@ -67,9 +67,21 @@ public abstract class SHA3 extends DigestBase {
         0x8000000000008080L, 0x80000001L, 0x8000000080008008L,
     };
 
+    // the starting 3 or 5 bits of the domain separator and 10*1 padding
+    // (11111... for the SHAKE functions and 011... for the SHA3-nnn ones)
     private final byte suffix;
+
+    // the stae matrix flattened into an array
     private long[] state = new long[DM*DM];
-    private int squeezeOffset = -1;
+
+    // The byte offset in the state where the next sqeeze() will start.
+    // -1 indicates that either we are in the absorbing phase (only
```

Comment:
I see `either` but no `or`. Did you meant to talk about `reset`?

Reply:
Fixed.